pilot/const/function_calls.py

```diff
@@ -200,29 +200,48 @@ def command_definition(description_command='A single command that needs to be ex
+                                    'enum': ['command', 'save_file', 'modify_file', 'human_intervention'],
                                     'description': 'Type of the development step that needs to be done to complete the entire task.',
                                 },
                                 'command': command_definition(),
-                                'code_change': {
+                                'save_file': {
                                     'type': 'object',
-                                    'description': 'A code change that needs to be implemented. This should be used only if the task is of a type "code_change".',
+                                    'description': 'A file that needs to be created or updated (completely replaced). This should be used for new files, small files (up to 100 lines), and files with where most of the content is replaced.',
                                     'properties': {
                                         'name': {
                                             'type': 'string',
-                                            'description': 'Name of the file that needs to be implemented.',
+                                            'description': 'Name of the file that needs to be created or replaced.',
                                         },
                                         'path': {
                                             'type': 'string',
-                                            'description': 'Full path of the file with the file name that needs to be implemented.',
+                                            'description': 'Full path of the file (with the file name) that needs to be created or replaced.',
                                         },
                                         'content': {
                                             'type': 'string',
-                                            'description': 'Full content of the file that needs to be implemented. **IMPORTANT**When you want to add a comment that tells the user to add the previous implementation at that place, make sure that the comment starts with `[OLD CODE]` and add a description of what old code should be inserted here. For example, `[OLD CODE] Login route`.',
+                                            'description': 'Full content of the file that needs to be implemented. Remember, you MUST NOT omit any of the content that should go into this file.',
                                         },
                                     },
                                     'required': ['name', 'path', 'content'],
                                 },
+                                'modify_file': {
+                                    'type': 'object',
+                                    'description': 'A file that should be modified. This should only be used for large existing files (more than 100 lines of code).',
```

Comment:
My plea to LLM to use this only for larger files fell on deaf ears (I guess that's because LLM has no ears?). It largery gets ignored. No matter, I feel this still should make it more likely to consider so I left it in there.

Reply:
My findings are that any time LLM has options which are not very clear (grey zone) he will in most cases stick to one. I think these descriptions go to grey zone and here are my suggestions what to try different:

First, save_file.description from:
`'A file that needs to be created or updated (completely replaced). This should be used for new files, small files (up to 100 lines), and files with where most of the content is replaced.'`
change to:
`'A file that needs to be created or file that needs to be completely replaced. This should be used for new files, files up to 100 lines, and files with where most of the content is replaced.'`

'most of the content' is still grey zone if you ask me, so maybe adding 'more than 90% of content is replaced' or something like that will make it more clear.

Second one would be modify_file.description from:
`'A file that should be modified. This should only be used for large existing files (more than 100 lines of code).'`
change to:
`'A file that should be modified. This should only be used for existing files that have more than 100 lines of code.'`

Just a suggestion, but I would always try to give as clear instructions as possible.